src/components/InputPhone/index.tsx

```diff
@@ -64,14 +65,23 @@ const InputPhone: React.FC<Props> = props => {
         setIsOpen(false);
     };
 
-    const handleChange = (phoneValue: string) => {
+    const handleChange = (phoneNumberValue: string = '') => {
+        setPhoneValue(phoneNumberValue);
         onChange({
             countryCode: selectedCountry.countryCode,
             isoCode: selectedCountry.isoCode,
-            phone: phoneValue,
+            phone: phoneNumberValue,
         });
     };
 
+    React.useEffect(() => {
+        onChange({
+            countryCode: selectedCountry.countryCode,
+            isoCode: selectedCountry.isoCode,
+            phone: phoneValue,
+        });
+    }, [selectedCountry, phoneValue, onChange]);
```

Comment:
This is currently causing an infinite loop when using it as controlled.
I think the only you need to do is fire onChange in selectCountry function and this will send the right value when the country is changed.

Reply:
Done, the "onChange" as a dependency in the effect array is the one causing the loop, if I removed it, all is fine but as you mentioned moving the onChange prop method inside the "selectCountry" function serves the same purpose and the effect is not need it.